### PR TITLE
fix(schema-bind): handle reserved keywords for serde in rust

### PR DESCRIPTION
**Pull request by Niraj-Kamdar**
Currently, Rust JSON serializer doesn't support de/serializing the reserved words. This PR fixes it.

- Close: https://github.com/polywrap/toolchain/issues/1221

### CODE REVIEW

**Review by cbrzn — COMMENTED** [review]
Would it make sense to add a test in the `test-cases/cases/wasm/wasm-rs/json-type` folder to ensure this works?

File: packages/js/client/src/__tests__/e2e/test-cases.ts [hunk 1]
```diff
@@ -473,6 +473,28 @@ export const runJsonTypeTest = async (client: PolywrapClient, uri: string) => {
 
   const methodJSONResult = JSON.stringify(json);
   expect(methodJSONResponse.data).toEqual(methodJSONResult);
+
+  if (testReserved) {
```
Comment (cbrzn) [inline packages/js/client/src/__tests__/e2e/test-cases.ts:477]: If this is added, then the `wasm-rs` file should be edited to pass `true` for this arg when calling the `runJsonTypeTest` function

File: packages/js/client/src/__tests__/e2e/test-cases.ts [hunk 2]
```diff
@@ -412,7 +412,7 @@ export const runInvalidTypesTest = async (
   );
 };
 
-export const runJsonTypeTest = async (client: PolywrapClient, uri: string) => {
+export const runJsonTypeTest = async (client: PolywrapClient, uri: string, testReserved: boolean = false) => {
```
Comment (dOrgJelli) [inline packages/js/client/src/__tests__/e2e/test-cases.ts:415]: Can you please pass in `true` for `testReserved` in the wasm-rs tests? This logic is not currently being used.

**Review by dOrgJelli — CHANGES_REQUESTED** [review]
LGTM!

**Review by cbrzn — APPROVED** [review]

**Review by dOrgJelli — APPROVED** [review]

**Pull request merged**